### ISSUE TITLE
Kirby V3 Update?

### ISSUE THREAD

**Issue by carstengrimm** [sebsel/seblog-kirby-webmentions#6]
Currently being on the way to a stable release kirby has published several nighty betas for dev's and other enthusiasts to try and play with the upcoming release of kirby.

I've been very interested in webmentions since the very beginning and even though most site users have never heared or even used it here, i would happily continue using webmentions in kirby 3 as well.

Is there any plans of updating or a new branch for v3?

**Comment by sebsel** [comment]
I haven't been following Kirby v3 at all for over a year now. I have no idea what changed. My own site still runs on Kirby v2, but I'm actually planning on moving away from it. (If only I had time.) This plugin is part of why I am moving away: in v2 there are not that much hooks for doing a thing like Webmention, and I ran into all kinds of limitations in both the toolkit and v2.

That said, I should give v3 a chance, I think, or at least check if the things I ran into have been changed.

To be specific:
- HTTP headers are stored in an PHP associative array. This means that in case of multiple `Link` headers, only the last one will survive, making it hard to parse endpoints.
- I needed some more URL parsing, for which I opened a PR, which was never really taken serious. (To be fair, I should've just included it in this plugin, which I later [did](https://github.com/sebsel/seblog-kirby-webmentions/commit/32c07e1994a1ded70d54951aa45a6ecd0753238f))
- The router in v2 only really supports routing the current request. I had to make a weird hack to overwrite the POST method from the webmention endpoint into a GET method for webmention lookup. [See here](https://github.com/sebsel/seblog-kirby-webmentions/blob/master/lib/endpoint.php#L100)

All in all it did feel to me like Kirby was a nice CMS, but just not that great of a web-framework.

**Comment by carstengrimm** [comment]
thanks for the heads up.

well kirby v3 is still in (more or less) closed beta, so every early contributor (purchasing a license) has access to the latest betas as well as their v3 documentations.

as far as what i can answer,
- v3 will have more extended hook methods such as events happening "before" and "after" there might be many more hooks in addition to those which are found in v2. v3 will also have some kind of api endpoints which can be called (maybe that could be useful?)
- the toolkit itself is revamped entirely... i am not sure if there's much new in addition

if this plugin were to be discontinued, what would be your guess is a good alternative to embed webmentions?

**Comment by mauricerenck** [comment]
@carstengrimm did you find any good alternatives? I am searching the web right now and it doesn't look good. I am using brid.gy and webmention.io but I also want to send mentions, which isn't working with these services alone.

I would start writing a plugin for K3 but that might take to long, I don't have enough time…

**Comment by carstengrimm** [comment]
No. It's not widespread and except big geeks will not even implement required markup on their site. To link back when writing an article. So I think it's a waste of time. I like mentions but that side it's not like anyone would pay extra for that. 

Therefore I have stopped looking for any sort of Alternative.

**Comment by sebsel** [comment]
I agree, support for Webmentions is not widespread. I want to point out, however, that the required markup (usually Microformats2) is only needed if you want to display received Webmentions. The receiving part just expects any source with HTML with a link to the target page. You can display received Webmentions also without the rich markup of Microformats2 (it does add value, but it is not part of the actual Webmention).

For me, the main part of IndieWeb (where Webmentions originate from) is posting content on my own site. The notifications about that content is secondary to me. But, I do still get a lot of joy seeing a like show up on a supporting site. I'm glad my site supports it at the moment (through this otherwise not so great plugin).

For sending Webmentions you could check out [Telegraph](https://telegraph.p3k.io/), which is a service (from the maker of Webmentions.io) that does just that.

I would also love to maintain this plugin better, but I have the same problem: not enough time.

**Comment by mauricerenck** [comment]
Maybe I will try to port it to K3 when I switch. I just got it running on K2 and can now send tweets, replys and stuff like that from my site, which I like to see as my base for all online activity. 

I think wordpress comes with webmentions or at least pingback built right in, so it might be spread more than one might think but most people aren't aware of it… Maybe because of all the bad facebook stuff happening people start thinking about alternatives to it and stuble upon webmention. "Die Hoffnung stirb zuletzt."

**Comment by sebsel** [comment]
Just want to chime in to point at this blogpost by another Sebastian: https://sebastiangreger.net/2019/05/sendmention-commention-webmentions-for-kirby-3/ in which he points to his work on a Webmention plugin for Kirby 3.

I will not be writing one myself, so I will now close this issue. If there are updates from my side I will try and keep this information up to date, but do not expect Kirby 3 support in this repo :)
